internal/entitlement/events.go

```diff
@@ -39,32 +35,54 @@ func (e entitlementEvent) Validate() error {
 
 type EntitlementCreatedEvent entitlementEvent
 
-var entitlementCreatedEventSpec = spec.EventTypeSpec{
-	Subsystem: EventSubsystem,
-	Name:      entitlementCreatedEventName,
-	Version:   "v1",
-}
+var (
+	_ marshaler.Event = EntitlementCreatedEvent{}
 
-func (e EntitlementCreatedEvent) Spec() *spec.EventTypeSpec {
-	return &entitlementCreatedEventSpec
-}
+	entitlementCreatedEventName = metadata.GetEventName(metadata.EventType{
+		Subsystem: EventSubsystem,
+		Name:      "entitlement.created",
+		Version:   "v1",
+	})
```

Comment:
I still think these should be reduced to a single string identifier, but it's not a big deal for now.